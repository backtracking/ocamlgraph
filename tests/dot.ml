(**************************************************************************)
(*                                                                        *)
(*  Ocamlgraph: a generic graph library for OCaml                         *)
(*  Copyright (C) 2004-2007                                               *)
(*  Sylvain Conchon, Jean-Christophe Filliatre and Julien Signoles        *)
(*                                                                        *)
(*  This software is free software; you can redistribute it and/or        *)
(*  modify it under the terms of the GNU Library General Public           *)
(*  License version 2, with the special exception on linking              *)
(*  described in file LICENSE.                                            *)
(*                                                                        *)
(*  This software is distributed in the hope that it will be useful,      *)
(*  but WITHOUT ANY WARRANTY; without even the implied warranty of        *)
(*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.                  *)
(*                                                                        *)
(**************************************************************************)

(* $Id:$ *)

open Graph
module G = Imperative.Digraph.Abstract(String)
module B = Builder.I(G)
module DotInput = 
  Dot.Parse
    (B)
    (struct 
      let node (id,_) _ = match id with
	| Dot_ast.Ident s
	| Dot_ast.Number s
	| Dot_ast.String s
	| Dot_ast.Html s -> s
      let edge _ = ()
    end)
module Display = struct
  include G
  let vertex_name v = "\"" ^ String.escaped (V.label v) ^ "\""
  let graph_attributes _ = []
  let default_vertex_attributes _ = []
  let vertex_attributes _ = []
  let default_edge_attributes _ = []
  let edge_attributes _ = [ `HtmlLabel "f&#36;oo"; `Tooltip "description"  ]
  let get_subgraph _ = None
end
module DotOutput = Graphviz.Dot(Display)

let g = DotInput.parse Sys.argv.(1)

let () = 
  Printf.printf "========= BEGIN output graph =========\n";
  DotOutput.output_graph stdout g;
  Printf.printf "========= END output graph ========="