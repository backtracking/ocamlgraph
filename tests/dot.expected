========= BEGIN output graph =========
digraph G {
  "task_menu";
  "task_ingredients";
  "task_invitation";
  "task_cook";
  "task_table";
  "task_eat";
  
  
  "task_menu" -> "task_ingredients" [tooltip="description", label=<f&#36;oo>, ];
  "task_ingredients" -> "task_cook" [tooltip="description", label=<f&#36;oo>, ];
  "task_invitation" -> "task_cook" [tooltip="description", label=<f&#36;oo>, ];
  "task_cook" -> "task_eat" [tooltip="description", label=<f&#36;oo>, ];
  "task_table" -> "task_eat" [tooltip="description", label=<f&#36;oo>, ];
  
  }
========= END output graph =========